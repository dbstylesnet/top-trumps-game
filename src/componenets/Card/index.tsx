import React from 'react'
import CardStyles from './styles'   

interface PlayerProps{
    playerITurn?: boolean, 
    cardType: string,
    attribute: string,
    attrValue: number,
    name: string,
    id?: string,
    player: number,
    winner?: string,
    playerIWins?: (x: any) => any,
    playerIIWins?: (x: any) => any,
    funcThatUpdates?: (x: any) => any,
}

const Card = (props: PlayerProps) => {
    return <CardStyles>
        <span onClick={props.player === 1 ? props.playerIWins : props.playerIIWins}>
            Player {props.player === 1 ? 'I' : 'II'}
            <div></div>
            Name: {props.name}
            <div></div>
            {props.attribute}: 
            {props.playerITurn && props.player === 1 ? 
                props.attrValue : 
                    !props.playerITurn && props.player === 2 ? 
                        props.attrValue :
                        '?'
            }
        </span>
    </CardStyles>
}

export default Card
