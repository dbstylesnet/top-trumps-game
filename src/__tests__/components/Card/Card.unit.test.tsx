import React from 'react'
import { shallow } from 'enzyme'
import Card from '../../../componenets/Card'

let gameState = {
    scores: [],
    scoreI: 0,
    scoreII: 0,
    isPlayerITurn: true,
    isTurnStarted: false,
    starships: [
        {
            id: "cj0nwtqoyq4tp0114jjljap6j",
            name: "Sentinel-class landing craft",
            hyperdriveRating: 1,
        },
        {
            id: "cj0nwtqphq4tr0114zo7suw8h",
            name: "Death Star",
            hyperdriveRating: 4,
        },
    ],
    persons: [],
}

// cardProps: isTurnStarted: true, playerITurn: true,
const cardProps = {
    isTurnStarted: true,
    playerITurn: true,
    cardType: "Starship",
    attribute: "hyperdriveRating",
    attrValue: 1,
    name: "Death Star",
    player: 1,
    onCardClick: () => {
        gameState = {
            ...gameState,
        }
    }
}


it('shallow renders without crashing', () => {
    const wrapper = shallow(<Card {...cardProps} />)
    expect(wrapper).toMatchSnapshot() 
    expect(wrapper.prop('className')).toEqual('card')
    expect(wrapper.containsMatchingElement(<span>Name: Death Star</span>)).toBeTruthy()
})

it('shallow renders with turnStarted and playerITurn', () => {
    const wrapper = shallow(<Card {...cardProps} />)
    expect(wrapper.prop('onClick')).toBeTruthy()
    expect(wrapper.containsMatchingElement(<span>hyperdriveRating:?</span>)).toBeTruthy() 
})


// cardProps: isTurnStarted: false, playerITurn: true,
const cardProps2 = {
    isTurnStarted: false,
    playerITurn: true,
    cardType: "Starship",
    attribute: "hyperdriveRating",
    attrValue: 1,
    name: "Death Star",
    player: 1,
    onCardClick: () => {
        gameState = {
            ...gameState,
        }
    }
}

it('shallow renders without turnStarted and with playerITurn', () => {
    const wrapper = shallow(<Card {...cardProps2} />)
    expect(wrapper.prop('onClick')).toBeFalsy()
    expect(wrapper.containsMatchingElement(<span>hyperdriveRating:1</span>)).toBeTruthy() 
})

// cardProps: isTurnStarted: true, playerITurn: false,
const cardProps3 = {
    isTurnStarted: true,
    playerITurn: false,
    cardType: "Starship",
    attribute: "hyperdriveRating",
    attrValue: 1,
    name: "Death Star",
    player: 1,
    onCardClick: () => {
        gameState = {
            ...gameState,
        }
    }
}

it('shallow renders without turnStarted and with playerITurn', () => {
    const wrapper = shallow(<Card {...cardProps3} />)
    expect(wrapper.prop('onClick')).toBeTruthy()
    expect(wrapper.containsMatchingElement(<span>hyperdriveRating:1</span>)).toBeTruthy() 
})

// cardProps: isTurnStarted: false, playerITurn: false,
const cardProps4 = {
    isTurnStarted: false,
    playerITurn: false,
    cardType: "Starship",
    attribute: "hyperdriveRating",
    attrValue: 1,
    name: "Death Star",
    player: 1,
    onCardClick: () => {
        gameState = {
            ...gameState,
        }
    }
}

it('shallow renders without turnStarted and with playerITurn', () => {
    const wrapper = shallow(<Card {...cardProps4} />)
    expect(wrapper.prop('onClick')).toBeFalsy()
    expect(wrapper.containsMatchingElement(<span>hyperdriveRating:1</span>)).toBeTruthy() 
})
