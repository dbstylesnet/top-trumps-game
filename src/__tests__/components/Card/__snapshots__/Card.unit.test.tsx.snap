// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow renders with turnStarted and playerITurn 1`] = `ShallowWrapper {}`;

exports[`shallow renders with turnStarted and playerITurn 2`] = `ShallowWrapper {}`;

exports[`shallow renders with turntarted and playerITurn 1`] = `ShallowWrapper {}`;

exports[`shallow renders without crashing 1`] = `ShallowWrapper {}`;

exports[`shallow renders without crashing wit 1`] = `ShallowWrapper {}`;

exports[`shallow renders without crashing with  1`] = `ShallowWrapper {}`;

exports[`shallow renders without crashing with turn started and  1`] = `ShallowWrapper {}`;

exports[`shallow renders without crashing with turn started and player 1`] = `ShallowWrapper {}`;

exports[`shallow renders without crashing with turn started and playerITurn 1`] = `ShallowWrapper {}`;

exports[`shallow renders without crashing with turn started and playerITurn 2`] = `ShallowWrapper {}`;

exports[`shallow renders without turnStarted and playerITurn 1`] = `ShallowWrapper {}`;

exports[`shallow renders without turnStarted and with playerITurn 1`] = `ShallowWrapper {}`;

exports[`shallow renders withstarted and playerITurn 1`] = `ShallowWrapper {}`;
